File: AppPy/model/NaivStandard.py
#Algoritmo basado en el libro METHODS OF MATRIX MULTIPLICATION AN OVERVIEW OF SEVERAL METHODS AND THEIR IMPLEMENTATION y apoyado en IA

def naiv_standard(matrizA, matrizB, matrizResultado, cantidadFilasMatrices, cantidadColumnasMatrices, cantidadMaximaIteracionesFilaColumna):
    
    aux = 0.0
    # Realiza la multiplicación de matrices
    for i in range(cantidadFilasMatrices):
        for j in range(cantidadColumnasMatrices):
            aux = 0.0
            # Realiza la multiplicación de elementos y suma los resultados
            for k in range(cantidadMaximaIteracionesFilaColumna):
                aux += matrizA[i][k] * matrizB[k][j]
            matrizResultado[i][j] = aux
            
    return matrizResultado

File: AppCs/AppCs/Algoritmos/NaivStandard.cs
//Algoritmo basado en el libro METHODS OF MATRIX MULTIPLICATION AN OVERVIEW OF SEVERAL METHODS AND THEIR IMPLEMENTATION y apoyado en IA
using System;

public static class NaivStandard
{
    /// <summary>
    /// Realiza la multiplicación de matrices utilizando el algoritmo estándar (naive).
    /// </summary>
    /// <param name="matrizA">La primera matriz a multiplicar.</param>
    /// <param name="matrizB">La segunda matriz a multiplicar.</param>
    /// <returns>La matriz resultado de la multiplicación.</returns>
    public static long[][] NaivStandardMultiply(long[][] matrizA, long[][] matrizB)
    {
        int cantidadFilasMatrizA = matrizA.Length;
        int cantidadColumnasMatrizA = matrizA[0].Length;
        int cantidadFilasMatrizB = matrizB.Length;
        int cantidadColumnasMatrizB = matrizB[0].Length;

        if (cantidadColumnasMatrizA != cantidadFilasMatrizB)
        {
            throw new ArgumentException("Las dimensiones de las matrices no son compatibles para la multiplicación.");
        }

        long[][] matrizResultado = new long[cantidadFilasMatrizA][];
        for (int i = 0; i < cantidadFilasMatrizA; i++)
        {
            matrizResultado[i] = new long[cantidadColumnasMatrizB];
            for (int j = 0; j < cantidadColumnasMatrizB; j++)
            {
                double aux = 0.0;
                for (int k = 0; k < cantidadColumnasMatrizA; k++)
                {
                    aux += matrizA[i][k] * matrizB[k][j];
                }
                matrizResultado[i][j] = (int)aux;
            }
        }
        return matrizResultado;
    }
}
